Hmwk_#3_Hangman/hangman_game.py

```diff
@@ -0,0 +1,159 @@
+import functools
+import random
+from functools import reduce
+import time
+import hangman_graphics
+
+#get data from file but not read so later it can be accesed twice without reopening
+def get_words(filename):
+    text = open(filename, "r", encoding="utf-8")
+
+    return text
+
+#define mode PC vs. PC, Player vs. PC
+def set_game_mode():
+    game_mode = input("What type of game you wanna play? (Manual / Auto) ").lower()
+    
+    if game_mode == "manual":
+        print("Manual game mode requires to interact with PC !")
+    elif game_mode == "auto":
+        print("Auto game mode allows just to watch the flow")
+
+    return game_mode
+
+#counter for PC tries to guess word
+def guess_counter(guesses):
+
+    if guesses == 9:
+        print("You have last try left...!")
+    elif guesses == 10:
+        print("Out of tries, you've lost!")
+
+    return guesses
+
+#get word to guess from file randomly or define it manually
+def choose_word(text):
+
+    words = text
+    randomize = input("Do you want to choose word by yourself or select a word randmoly? (Choose / Random)...").lower()
+    wordlist = [word.lower().strip() for word in words.readlines()]
+
+    if randomize == 'choose':
+        word_to_guess =  input("Choose the word from to be guessed from a list")
+    elif randomize == 'random':
+        word_to_guess = random.choice(wordlist)
+
+    return word_to_guess
+
+#function to be used in 'reduce' to count characters frequency on fly
+def counter(letter_freq, char):
+
+    if char.isalpha() == True:
```

Comment:
It would be better to write: 

`if char.isalpha():`

according to Python best practices: https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html